probe-rs/src/core/mod.rs

```diff
@@ -380,6 +385,7 @@ impl SpecificCoreState {
         state: &'probe mut CoreState,
         memory: Memory<'probe>,
         base_address: Option<u32>,
```

Comment:
Couldn't we store this info on the CoreState (even if it's just used once and does not change), to make this interface more concise?

Reply:
Yes, I ended up adding the CoreAccessOptions to CoreState, which should work nicely for non-ARM architectures as well.